pages/docs/reference/sealed-classes.md

```diff
@@ -7,38 +7,83 @@ title: "Sealed Classes"
 
 # Sealed Classes
 
-Sealed classes are used for representing restricted class hierarchies, when a value can have one of the types from a
-limited set, but cannot have any other type. They are, in a sense, an extension of enum classes: the set of values
-for an enum type is also restricted, but each enum constant exists only as a single instance, whereas a subclass
-of a sealed class can have multiple instances which can contain state.
+_Sealed_ classes represent restricted class hierarchies that provide more control over inheritance.
+All subclasses of a sealed class are known at compile time. No other subclasses may appear after
+a module with the sealed class is compiled. For example, third-party clients can't extend your sealed class in their code.
+Thus, each instance of a sealed class belongs a type from a limited set that is known when this class is compiled. 
```

Comment:
Seems like a preposition is missing in `belongs to`. However, `has a type` seems more natural to me.